### PR TITLE
Adding definition of plugins vs packages

### PR DESCRIPTION
**Pull request by sfshaza2**
As it says.

### CODE REVIEW

File: src/docs/development/packages-and-plugins/using-packages.md [hunk 1]
```diff
@@ -7,6 +7,30 @@ Flutter supports using shared packages contributed by other developers
...
+  package&mdash;the full designation is "plugin package",
+  which is generally shortened to "plugin".
+
+  **Packages**
+  : A package contains shared and reusable code and includes a list of
+    dependencies (specified in the `pubspec.yml` file), and (optionally)
+    Dart libraries, apps, resources, tests, images, and examples.
+    The [pub][] site lists many packages developed by Google engineers
+    and other generous members of the Flutter and Dart community
+    that you can use in your app.
```
Comment (kwalrath) [inline src/docs/development/packages-and-plugins/using-packages.md:22]: I found this a little hard to read, plus it seemed weird to imply that Google engineers are all generous. :) But maybe it should be "by Google engineers and other members..."

```suggestion
    The [pub.dev][] site lists many packages—developed by Google engineers
    and generous members of the Flutter and Dart community—
    that you can use in your app.
```

(and change [pub] to [pub.dev] below)

File: src/docs/development/packages-and-plugins/using-packages.md [hunk 2]
```diff
@@ -7,6 +7,30 @@ Flutter supports using shared packages contributed by other developers
...
+
+  **Plugins**
+  : A plugin package is a special kind of package that makes
+    platform functionality available to the app.
+    Plugin packages can be written for Android (using Kotlin or Java),
+    iOS (using Swift or Objective-C), web, macOS, Windows, or Linux,
```
Comment (kwalrath) [inline src/docs/development/packages-and-plugins/using-packages.md:28]:
```suggestion
    iOS (using Swift or Objective-C), web, macOS, Windows, Linux,
```

File: src/docs/development/packages-and-plugins/using-packages.md [hunk 3]
```diff
@@ -7,6 +7,30 @@ Flutter supports using shared packages contributed by other developers
...
+    or any combination thereof.
+    An example of a plugin might provide the Flutter app
```
Comment (kwalrath) [inline src/docs/development/packages-and-plugins/using-packages.md:30]:
```suggestion
    For example, a plugin might provide Flutter apps
```

File: src/docs/development/packages-and-plugins/using-packages.md [hunk 4]
```diff
@@ -27,21 +51,21 @@ existing published packages.
 
 ### Searching for packages
 
-Packages are published to [pub.dev][].
+Packages are published to the [pub][] site.
```
Comment (kwalrath) [inline src/docs/development/packages-and-plugins/using-packages.md:54]: We should talk about this terminology... it doesn't match what go/dart-terms currently says.

File: src/docs/development/packages-and-plugins/using-packages.md [hunk 5]
```diff
@@ -7,6 +7,30 @@ Flutter supports using shared packages contributed by other developers
 to the Flutter and Dart ecosystems. This allows quickly building
 an app without having to develop everything from scratch.
 
+{{site.alert.secondary}}
+  **What is the difference between a Flutter package
```
Comment (sfshaza2) [inline src/docs/development/packages-and-plugins/using-packages.md:11]:
```suggestion
  **What is the difference between a package
```

File: src/docs/development/packages-and-plugins/using-packages.md [hunk 6]
```diff
@@ -7,6 +7,30 @@ Flutter supports using shared packages contributed by other developers
...
+  and a plugin?** Actually, a plugin is a _type_ of
+  package&mdash;the full designation is "plugin package",
+  which is generally shortened to "plugin".
```
Comment (sfshaza2) [inline src/docs/development/packages-and-plugins/using-packages.md:14]:
```suggestion
  and a plugin?** Actually, a plugin is a _type_ of
  package&mdash;the full designation is _plugin package_,
  which is generally shortened to _plugin_.
```

File: src/docs/development/packages-and-plugins/using-packages.md [hunk 7]
```diff
@@ -7,6 +7,30 @@ Flutter supports using shared packages contributed by other developers
...
+{{site.alert.secondary}}
+  **What is the difference between a package
+  and a plugin?** Actually, a plugin is a _type_ of
+  package&mdash;the full designation is _plugin package_,
+  which is generally shortened to _plugin_.
+
+  **Packages**
+  : A package contains shared and reusable code and includes a list of
+    dependencies (specified in the `pubspec.yml` file), and (optionally)
+    Dart libraries, apps, resources, tests, images, and examples.
```
Comment (sfshaza2) [inline src/docs/development/packages-and-plugins/using-packages.md:19]:
```suggestion
    : At a minimum, a Dart package is a directory
    containing a pubspec file. Additionally,
    a package can contain dependencies
    (listed in the pubspec), Dart libraries, apps,
    resources, tests, images, and examples.
```

**Review by sfshaza2 — COMMENTED** [review]
Boy, what a mess this new suggestion feature can be. It's hard for me to tell if the suggestions that I made are correct on top of your suggestions.

**Comment by kwalrath** [comment]
> Boy, what a mess this new suggestion feature can be. It's hard for me to tell if the suggestions that I made are correct on top of your suggestions.

You can batch suggestions — though only from the files tab. That makes things a little less messy. I'll take another look if you're ready.

**Review by kwalrath — COMMENTED** [review]
LGTM once you update the pub site references

File: src/docs/development/packages-and-plugins/using-packages.md [hunk 8]
```diff
@@ -7,6 +7,32 @@ Flutter supports using shared packages contributed by other developers
...
+
+  **Packages**
+    : At a minimum, a Dart package is a directory
+    containing a pubspec file. Additionally,
+    a package can contain dependencies
+    (listed in the pubspec), Dart libraries, apps,
+    resources, tests, images, and examples.
+    The [pub.dev][] site lists many packages—developed by Google engineers
+    and generous members of the Flutter and Dart community—
+    that you can use in your app.
```
Comment (sfshaza2) [inline src/docs/development/packages-and-plugins/using-packages.md:24]:
```suggestion
    The [pub][] site lists many packages—developed by Google engineers
    and generous members of the Flutter and Dart community—
    that you can use in your app.
```

**Review by sfshaza2 — COMMENTED** [review]
@kwalrath, I believe everything has been modified, please check.

**Comment by sfshaza2** [comment]
I came THIS CLOSE to just creating a new PR.

**Comment by sfshaza2** [comment]
It's not letting me commit any suggestions, so I don't know how to make that happen.

**Comment by kwalrath** [comment]
> It's not letting me commit any suggestions, so I don't know how to make that happen.

I can do it.

**Comment by kwalrath** [comment]
Staged: https://kw-flutter-1.firebaseapp.com/docs/development/packages-and-plugins/using-packages

**Pull request merged**